config/proto/targets.proto

```diff
@@ -0,0 +1,44 @@
+/*
+  This proto defines the config for a Target. 
+  Targets are used by Hermes and Cloudprober to identify storage systems to monitor.
+*/
+syntax = "proto3";
+
+package hermes;
+
+option go_package = "github.com/googleinterns/step224-2020/targets/proto";
+
+// Definition of a Target, a storage system to be probed by a specific probe.
+message TargetsDef {
+  enum TargetSystem {
+    GOOGLE_CLOUD_STORAGE = 0;
+    CEPH = 1;
+  }
+
+  enum ConnectionType {
+     HTTP = 0;
+     HTTPS = 1;
+  }
+
+  TargetSystem target_system = 1;
+  ConnectionType connection_type = 2;
+  ConnectionDetails connection_details = 3;
+  int64 total_space_allocated = 4; // in MBs.
+}
+
+// Necessary when connecting to an API (e.g. Ceph).
+// Unclear yet if we need this for GCS, as we will use the Go client library and service account credentials.
+message ConnectionDetails {
```

Comment:
If most of these configs will only be relevant for Ceph, then take the target URL out and make it part of the TargetsDef and then rename this message to something Ceph specific (e.g. CephConnectionDetails), then make the CephConnectionDetails an optional (in proto3 syntax/style friendly) field.

Reply:
I have removed target_url and placed it in the TargetsDef message.

In proto3, all fields are optional (no enforcement), so there isn't any special syntax required.
```suggestion
message CephConnectionDetails {
```